Tests/EventsExporter/Persistence/FileWriterTests.swift

```diff
@@ -97,29 +101,35 @@ class FileWriterTests: XCTestCase {
             )
         )
 
-        let ioInterruptionQueue = DispatchQueue(label: "com.datadohq.file-writer-random-io")
+        let ioInterruptionQueue = DispatchQueue(
+            label: "com.datadohq.file-writer-random-io",
+            target: .global(qos: writer.queue.qos.qosClass)
+        )
 
         func randomlyInterruptIO(for file: File?) {
-            ioInterruptionQueue.async { try? file?.makeReadonly() }
-            ioInterruptionQueue.async { try? file?.makeReadWrite() }
+            ioInterruptionQueue.async {
+                try? file?.makeReadonly()
+                usleep(300)
+                try? file?.makeReadWrite()
+            }
         }
```

Comment:
Behaviour here is different from the previous code, is that intentional? I guess it's about the flakiness?

Reply:
It does the same job, but in a slightly different way. It breaks writes better (because of the bigger read-only timeframe) plus decreased amount of queue switches (which helped with flakiness on the GitHub, but never resolved it).